lib/elixir/src/elixir_def.erl

```diff
@@ -419,7 +419,11 @@ format_error({invalid_def, Kind, NameAndArgs}) ->
   io_lib:format("invalid syntax in ~ts ~ts", [Kind, 'Elixir.Macro':to_string(NameAndArgs)]);
 
 format_error(invalid_args_for_bodyless_clause) ->
-  "can use only variables and \\\\ as arguments in definition header";
+  "only variables and \\\\ are allowed as arguments in definition header.\n"
+  "\n"
+  "If you did not intend to define a header, make sure your function definition has\n"
+  "the proper syntax by wrapping the arguments in parentheses and ensuring there\n"
+  "is no space between the function name and arguments.\n";
```

Comment:
We should not introduce the new lines here because the formatting can be totally off depending on the terminal width. Please remove `\n` from the three lines above.